### PR TITLE
Added initial items command implementation

### PR DESCRIPTION
**Pull request by nightshadows**
- read item cards from "items" cards subdir
- added new "items" command with "hand", "add" and "discard" subcommands
- fixed styling via "black"

### CODE REVIEW

File: bot_commands.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-cards - (player>) [all|show|draw|discard|flip|hand] <card>
+cards - (player) [all|show|draw|discard|flip|hand] <card>
+items - (player) [(hand)|add|discard] <item>
```
Comment (vit-ka) [inline bot_commands.txt:2]: What do () mean?

File: d20potz.py [hunk 2]
```diff
@@ -19,7 +19,8 @@
 from potz import roll20
 
 D20PotzBotConfiguration = collections.namedtuple(
-    "D20PotzBotConfiguration", "db_location token cards_dir spelling order hp_defaults xp_default"
+    "D20PotzBotConfiguration",
+    "db_location token cards_dir spelling order hp_defaults xp_default",
```
Comment (vit-ka) [inline d20potz.py:23]: :love:

File: d20potz.py [hunk 3]
```diff
@@ -209,27 +209,63 @@ def set_claim_status(chat_id, player_name, user_id, claim: bool):
...
-        
+
+
+def get_player_items(chat_id, player_id):
+    items_key = f"player_items_{chat_id}_{player_id}".encode("utf-8")
+    items = DB.Get(items_key, default="".encode("utf-8"))
+    return items.decode("utf-8").split()
+
+
+def add_player_item(chat_id, player_id, item_id) -> bool:
```
Comment (vit-ka) [inline d20potz.py:243]: nit: Add full typing, not just return type?

File: d20potz.py [hunk 4]
```diff
@@ -280,9 +316,9 @@ async def hp_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
     player_name = get_player_by_user(chat_id, update.message.from_user.id)
     params = update.message.text.split()
 
-    if len(params) > 1:
```
Comment (vit-ka) [inline d20potz.py:283]: omg, who accepted THAT?

File: d20potz.py [hunk 5]
```diff
@@ -596,21 +632,110 @@ async def cards_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
         )
         return
 
+
+async def items_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
+    params = update.message.text.split()
```
Comment (vit-ka) [inline d20potz.py:637]: RFC: should we properly react to edited messages too?

File: d20potz.py [hunk 6]
```diff
@@ -596,21 +632,110 @@ async def cards_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
...
+    chat_id = update.effective_chat.id
+
+    player_name = get_player_by_user(chat_id, update.message.from_user.id)
+
+    if len(params) > 1:
+        player_param = params[1].lower()
+        player_list = CONFIG.order.lower().split()
+        if player_param in player_list:
+            player_name = player_param
+            del params[1]
```
Comment (vit-ka) [inline d20potz.py:647]: :puke

File: d20potz.py [hunk 7]
```diff
@@ -596,21 +632,110 @@ async def cards_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
...
+            "items",
+            get_player_items(update.effective_chat.id, player_name),
+            chat_id=chat_id,
+            context=context,
+        )
+        return
+    elif sub_command == "add":
+        if len(params) < 2:
+            await context.bot.send_message(
+                chat_id=chat_id, text="Usage: /items add <item name> (player)"
```
Comment (vit-ka) [inline d20potz.py:661]: So, is the player at the beginning or at the end?

File: d20potz.py [hunk 8]
```diff
@@ -596,21 +632,110 @@ async def cards_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
...
+            )
+        else:
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="{} already has {}".format(
+                    spell_hero_name(player_name), item_cards[0]
+                ),
+            )
+        return
+    elif sub_command == "discard":
```
Comment (vit-ka) [inline d20potz.py:687]: retire!

File: d20potz.py [hunk 9]
```diff
@@ -596,21 +632,110 @@ async def cards_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
...
+                    spell_hero_name(player_name), item_cards[0]
+                ),
+            )
+        else:
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="{} did not have {}".format(
+                    spell_hero_name(player_name), item_cards[0]
+                ),
+            )
```
Comment (vit-ka) [inline d20potz.py:716]:
```suggestion
            await context.bot.send_message(
                chat_id=chat_id,
                text="{} {} {}".format(
                    spell_hero_name(player_name), "removed" if removed else "did not have", item_cards[0]
                ),
            )

```

**Review by vit-ka — COMMENTED** [review]
Up to you if you want to clean up a bit (especially copypasta I commented on) or merge.

File: d20potz.zip
Comment (bestander) [inline d20potz.zip:1]: what's this zip for?

**Review by bestander — APPROVED** [review]

**Pull request closed without merging**